### ISSUE TITLE
SSHD doesn't work.

### ISSUE THREAD

**Issue by MrSorcus** [NixOS/nixpkgs#26776]
### Steps to reproduce
1:
`cat ./custom.nix`
```
{ config, lib, pkgs, ... }:

{
  imports = [ ./modules/installer/cd-dvd/installation-cd-base.nix ];

        users.extraUsers.root.openssh.authorizedKeys.keys = [ "ssh-ed25519 AAAA... sorcus@laptop" ];

        systemd.network.enable = true;
        systemd.network.networks.wired.name = "enp0s3";
        systemd.network.networks.wired.address = [ "2a01:***:***::13/64" ];
        systemd.network.networks.wired.gateway = [ "fe80::1" ];

        services.openssh.enable = true;
}
```
2:
`nix-build -A config.system.build.isoImage -I nixos-config=./custom.nix ./default.nix`

3:
Create virtual machine and run with nixos iso. SSH Daemon doesn't work. It's loaded, but not active.

4:
But it's working with `services.openssh.startWhenNeeded = true;` ... and show this warnings:
```
Jun 22 21:19:09 nixos systemd[1]: sshd.service: Service lacks both ExecStart= and ExecStop= setting. Refusing.
Jun 22 21:07:44 nixos systemd[1]: Started Session 3 of user root.
Jun 22 21:07:50 nixos systemd[1]: sshd.service: Service lacks both ExecStart= and ExecStop= setting. Refusing.
```

## Technical details

* System: 17.09.git.c89efa3 (Hummingbird)
* Nix version: nix-env (Nix) 1.11.11
* Nixpkgs version: 17.09.git.c89efa3
* Sandboxing enabled: build-use-sandbox = false

**Comment by vcunat** [comment]
And why is it wrong?  It's even documented: http://nixos.org/nixos/manual/#sec-installation (7.)

**Comment by aszlig** [comment]
@vcunat: He provided `authorizedKeys`, so it should work without setting a root password. We even have [tests](https://github.com/NixOS/nixpkgs/blob/6fb9f892382b4b091fc9edcae00e2eb4c0729bda/nixos/tests/openssh.nix) doing exactly that, but without the installer CD `import`, so I'm guessing the real issue here could be the latter.

**Comment by aszlig** [comment]
The actual problem here is [this line](https://github.com/NixOS/nixpkgs/blob/6fb9f892382b4b091fc9edcae00e2eb4c0729bda/nixos/modules/profiles/installation-device.nix#L55), so @MrSorcus adding the following line in your configuration should fix that:

```
systemd.services.sshd.wantedBy = lib.mkOverride 40 [ "multi-user.target" ];
```

For the installer image it makes sense to remove `sshd` from `multi-user.target`, because it allows the user to optionally enable `openssh` at runtime via `systemctl start sshd`.

**Comment by vcunat** [comment]
To be clear, I meant the fact that you have to start sshd manually on an installation system.  I can see nothing wrong about that.  (Perhaps I misunderstood.)

**Comment by aszlig** [comment]
@vcunat: No, I was wrong about that, I only read point 7 regarding setting the root passwd.

**Comment by Mic92** [comment]
This example should work: https://github.com/nixos-users/wiki/wiki/Installation-guide#build-a-custom-installation-image

**Comment by Mic92** [comment]
The warnings when using `services.openssh.startWhenNeeded` can be ignored.

**Comment by MrSorcus** [comment]
@Mic92 , maybe not a bug, but an error.
```
[root@nixos:~]# systemctl status sshd
● sshd.service
   Loaded: error (Reason: Invalid argument)
   Active: inactive (dead)

Jun 23 09:55:46 nixos systemd[1]: sshd.service: Service lacks both ExecStart= and ExecStop= setting. Refusing.
```

**Comment by MrSorcus** [comment]
@aszlig , thank you.
`systemd.services.sshd.wantedBy = lib.mkOverride 40 [ "multi-user.target" ];` work very well without any problems.

**Comment by tomberek** [comment]
Close as answered?

**Comment by MrSorcus** [comment]
> Close as answered?

I don't know.
@Mic92 is it fixed in documentation?

**Comment by Mic92** [comment]
Probably not.

**Comment by zauberpony** [comment]
https://nixos.wiki/wiki/Creating_a_NixOS_live_CD works flawlessly.

**Comment by flokli** [comment]
A fix for this is in progress at #63773.
